FluentNest/Filters.cs

```diff
@@ -113,23 +113,28 @@ public static QueryContainer GenerateComparisonFilter<T>(this Expression express
             var binaryExpression = (BinaryExpression)expression;
 
             var value = GetValue(binaryExpression);
-            var fieldName = GetFieldNameFromMemberOrGetFieldNamed(binaryExpression.Left);
-            var filterDescriptor = new QueryContainerDescriptor<T>();
 
-            if (value is DateTime)
+            if (value == null)
             {
-                return filterDescriptor.DateRange(x => x.RangeOnDate(type, (DateTime)value).Field(fieldName));
+                // if the value is null, no filters are added
```

Comment:
There is no concept of null value ?

Reply:
This code is only for comparisons: I think it's ok that when comparing anything against null the code should return an empty filter to match all. But that's a good remark, because when it comes to equality ```null``` is already interpreted as a nonexistence equality filter elsewhere in the code (i.e. a filter where x == null is interpreted as a record with a [missing field](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-exists-query.html#_literal_missing_literal_query)). As an aside, I believe ```null_value``` [custom mappings](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-exists-query.html#_literal_null_value_literal_mapping) are not taken into account anywhere.